pycode/test/test_get_vaccination_data.py

```diff
@@ -0,0 +1,284 @@
+            ('2020-12-27', str(10001), '18-59', 3, 222),
+            ('2020-12-27', str(10001), '60+', 1, 22),
+            ('2020-12-27', str(10001), '60+', 2, 332),
+            ('2020-12-27', str(10001), '60+', 3, 76)
+        ]
+        df_to_split = pd.DataFrame(
+            vacc_data, columns=col_names_vacc_data)
+
+        column_ident = 'Impfschutz'
+        column_vals_name = 'Anzahl'
+        # wrong amount for data
+        new_col_labels = ['Vacc_partially', 'Vacc_completed']
+        df_split = gvd.split_column_based_on_values(
+            df_to_split, column_ident, column_vals_name, new_col_labels)
+
+        new_column_names1 = [
+            'Impfdatum',
+            'LandkreisId_Impfort',
+            'Altersgruppe',
+            'new_column_1']
+        new_column_names2 = [
+            'Impfdatum',
+            'LandkreisId_Impfort',
+            'Altersgruppe',
+            'new_column_2']
+        new_column_names3 = [
+            'Impfdatum',
+            'LandkreisId_Impfort',
+            'Altersgruppe',
+            'new_column_3']
+        new_column_names = []
+        new_column_names.append(new_column_names1)
+        new_column_names.append(new_column_names2)
+        new_column_names.append(new_column_names3)
+
+        indent_unique = df_to_split[column_ident].unique()
+        number_unique_idents_df = []
+        for i in range(0, len(indent_unique)):
+            number_unique_idents_df.append(
+                len(df_to_split[df_to_split[column_ident] == indent_unique[i]]))
+
+        self.assertEqual(
+            len(df_split),
+            3, 'Did not divide dataframe in three sperate ones.')
+        for i in range(0, len(new_col_labels)):
+            self.assertEqual(
+                number_unique_idents_df[i],
+                len(df_split[i]),
+                'Length of Dataframe is wrong.')
+            column_names_i = df_split[i].columns
+            for j in range(0, len(column_names_i)):
+                self.assertEqual(
+                    new_column_names[i][j],
+                    column_names_i[j],
+                    'Column names do not match.')
+
+    def test_intervall_mapping(self):
+        lower_bounds1 = np.array(
+            [0, 3, 6, 15, 18, 25, 30, 40, 50, 65, 74, 100])
+        upper_bounds1 = np.array(
+            [0, 3, 5, 6, 12, 15, 18, 25, 30, 35, 40, 50, 60, 65, 74, 80, 100])
+        lower_bounds2 = np.array([0, 10, 20, 70, 100])
+        upper_bounds2 = np.array([0, 5, 15, 20, 50, 60, 70, 85, 90, 100])
+
+        test_map1 = [
+            [[1, 0]],
+            [[2 / 3, 1], [1 / 3, 2]],
+            [[2 / 3, 3], [1 / 3, 4]],
+            [[1, 5]],
+            [[1, 6]],
+            [[1, 7]],
+            [[0.5, 8], [0.5, 9]],
+            [[1, 10]],
+            [[2/3, 11], [1/3, 12]],
+            [[1, 13]],
+            [[6/26, 14], [20/26, 15]]]
+
+        test_map2 = [
+            [[1 / 2, 0], [1 / 2, 1]],
+            [[1.0, 1], [0.0, 2]],
```

Comment:
If this is the return of my function `gvd.create_intervals_mapping`, then my function is wrong.

[10,20] should be mapped with [0.5,1] and [0.5,2]. 

Please get in touch with me.

Reply:
See issue #139 :-)